lab-taylor/server.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const http = require('http');
+const parseUrl = require('url').parse;
+const parseQuery = require('querystring').parse;
+const bodyParser = require('./lib/parse-req');
+const cowsay = require('cowsay');
+
+const port = process.argv[2] || 3000;
+const badReq = {text: 'bad request\ntry: localhost:3000/cowsay?text=howdy&face=ghostbusters', face:'ghostbusters'};
+
+function buildCowRes(options, status, res) {
+  res.writeHead(status, {
+    'Content-Type': 'text/plain'
+  });
+
+  const face = options.face || 'default';
+  res.write(cowsay.say({
+    text: options.text.toString(),
+    f: face
+  }));
+  res.end();
+}
+
+function defaultRes(res){
+  res.writeHead(200, {
```

Comment:
This should be a 404 rather than a 400 since it isn't that the request is bad (i.e. bad json or query) but that the endpoint isn't part of the api.